### DISCUSSION TITLE
Can we dynamically edit the values of existing variables or add new variables in the Supabase email template?

### DISCUSSION THREAD

**Discussion by siddheshmtechzillla** [supabase/supabase#7075] [Questions]
I was going through the [documentation](https://supabase.com/docs/gotrue/server/about) and I found that for an email there are three variables available by default under `signIn` API which are: 
1. `SiteURL`
2. `Email`
3. `ConfirmationURL`

I had a use case in my mind where I wanted to send a variable dynamically to the email template while using the `signIn` API.

The way I was thinking of two options:
1. Either edit the value of one of the variables while using the `signIn` API.
2. Or send a new variable while using the `signIn` API.

The known parameter to me is that I have to set this variable in the Supabase authentication template.

The unknown parameter to me is that: 
1. Can I send or edit a variable for the email template while using the  `signIn` API.
2. Use this variable in the Supabase authentication template. ( Note: This query only exists if we cannot edit the existing variable and add a new variable. )

It would be helpful for me to have any help/resources on the above issue.

**Comment by hdoro** [comment]
This isn't answering exactly what you want, Siddhesh, but perhaps if your variables aren't dyanmic on a per-user basis you can use conditionals to power your templates? Anywho, just sharing my learnings, which I've also published here - **[Using variables conditionally in Supabase’s transactional e-mail](https://mulungood.com/supabase-emails-conditional-variables)**:

---

If you have user metadata values you want to use in your email templates, but don’t want to risk having users receive `“Welcome to our app, NULL”` in the case where metadata is missing, you can use variables conditionally.

```html
<!-- In an ideal world... -->
<h1>Welcome to our app, {{ .Data.name || "" }}!</h1>
```

Supabase uses [gomail.v2](https://github.com/go-gomail/gomail) to send e-mails, which is powered by Go’s [html/template](https://pkg.go.dev/html/template). Providing default values for variables can be achieved with the `{{ or }}` function:

```html
<h1>Welcome to our app, {{ or .Data.name "" }}!</h1>
```

You can also use `if` statements for more advanced templates:

```html
{{ if .Data.favoriteColor }}
<p>
	We're are also fans of {{ .Data.favoriteColor }} -
	let's paint the world with it, {{ or .Data.name "together" }}!
</p>
{{ end }}
```

You can use other features of Go’s `html/template`, of course - it’s so powerful it’s even used by the [Hugo static site generator (SSG)](https://gohugo.io/) to create entire websites!
